src/trng/mod.rs

```diff
@@ -1 +1,40 @@
 pub mod registers;
+use crate::memory_map::mmio;
+use registers::Registers;
+
+pub struct TRNG {
+    registers: Registers,
+    pub has_key: bool,
```

Comment:
Maybe having this public could cause one to set `has_key` to *true* when the key was never generated. Maybe this should remain private, and a 'getter' is made to check the status of `has_key`.